packages/system/src/Item/DetailView/Tabs/General.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import {
+  DetailContainer,
+  DetailInputWithLabelRow,
+  useTranslation,
+  DetailSection,
+  Checkbox,
+  Grid,
+  NumericTextInput,
+} from '@openmsupply-client/common';
+import { useItem } from '../../api';
+
+export const GeneralTab = () => {
+  const t = useTranslation('catalogue');
+  const { data, isLoading } = useItem();
+  const isDisabled = true;
+
+  if (isLoading) return null;
+  return (
+    <DetailContainer>
+      <Grid
+        container
+        flex={1}
+        flexDirection="column"
+        style={{ maxWidth: 500 }}
+        gap={4}
+      >
+        <DetailSection title={t('title.details')}>
+          <DetailInputWithLabelRow
+            label={t('label.unit')}
+            inputProps={{ value: data?.unitName, disabled: isDisabled }}
+          />
+          <DetailInputWithLabelRow
+            label={t('label.strength')}
+            inputProps={{ value: data?.strength, disabled: isDisabled }}
+          />
+          <DetailInputWithLabelRow
+            label={t('label.ddd')}
+            inputProps={{ value: data?.ddd, disabled: isDisabled }}
+          />
+          <DetailInputWithLabelRow
+            label={t('label.type')}
+            inputProps={{ value: data?.type, disabled: isDisabled }}
+          />
+          <DetailInputWithLabelRow
+            label={t('label.doses')}
+            Input={
+              <NumericTextInput value={data?.doses} disabled={isDisabled} />
+            }
+          />
+          <DetailInputWithLabelRow
+            label={t('label.is-vaccine')}
+            inputProps={{ value: data?.isVaccine, disabled: isDisabled }}
+            Input={<Checkbox disabled={isDisabled} checked={data?.isVaccine} />}
+          />
+        </DetailSection>
+        <DetailSection title={t('title.categories')}>
+          <DetailInputWithLabelRow
+            label={t('label.atc-category')}
+            inputProps={{ value: data?.atcCategory, disabled: isDisabled }}
+          />
+          <DetailInputWithLabelRow
+            label={t('label.universal-name')}
+            inputProps={{
+              value: data?.msupplyUniversalName,
+              disabled: isDisabled,
+            }}
+          />
+          <DetailInputWithLabelRow
+            label={t('label.universal-code')}
+            inputProps={{
+              value: data?.msupplyUniversalCode,
+              disabled: isDisabled,
+            }}
+          />
+        </DetailSection>
```

Comment:
Would it be possible to generate these from the query response fields themselves rather than hard-coding each field? Iterate over the `data` object fields? Could even use the field name itself as the key for the language string (i.e. ``label={t(`label.${fieldName}`)}``)

Then we can use the same DetailView component for other Detail modals. And if the data fields from the back-end changes, we won't have to change the front-end.

I guess I'm suggesting a generalise Detail view component like the DataTable component is for List views.

Reply:
that is a good idea - not sure how to handle different input components, data validation or grouping of items though. that would require some mapping, like a configuration object and then it isn't completely dynamic.

I'm still favouring the declarative syntax for component use, so would keep this rather longhand version. We have planned to use JSON schema forms and this is great use case for that.